### PR TITLE
Create charcoal texture

### PR DESCRIPTION
**Pull request by sschuraytz**

### CODE REVIEW

File: app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java [hunk 1]
```diff
@@ -76,4 +85,26 @@ public void onDraw(Canvas canvas) {
     public Paint getPaint() {
         return paint;
     }
+
+    private void printTexturedCircle(float pointX, float pointY)
+    {
+        rand = new Random();
```
Comment (andrewoid) [inline app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java]: There is no reason to create Random each time (and you aren't using it).
Create it once.

File: app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java [hunk 2]
```diff
@@ -76,4 +85,26 @@ public void onDraw(Canvas canvas) {
...
+        int angle;
+        int magnitude;
+        float horizontalShift;
+        float verticalShift;
+        for (int i = 0; i < 30; i++)
+        {
+            angle = ThreadLocalRandom.current().nextInt(0, 360 + 1);
```
Comment (andrewoid) [inline app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java]: Use `rand` variable instead.

File: app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java [hunk 3]
```diff
@@ -44,9 +51,11 @@ public boolean onTouchEvent(MotionEvent event) {
         switch(event.getAction()) {
             case MotionEvent.ACTION_DOWN:
                 path.moveTo(pointX, pointY);
+                printTexturedCircle(pointX, pointY);
```
Comment (andrewoid) [inline app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java]: You are creating a new circle each time you draw it.
Instead, draw it to a separate bitmap once, then draw the bitmap to canvasBitmap each time you want to draw the circle.

**Review by andrewoid — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java [hunk 4]
```diff
@@ -44,17 +50,20 @@ public boolean onTouchEvent(MotionEvent event) {
 
         switch(event.getAction()) {
             case MotionEvent.ACTION_DOWN:
+                //TODO: need to be able to adjust circle size and opacity in this class
+                printCircleWithLocation(pointX, pointY);
                 path.moveTo(pointX, pointY);
```
Comment (andrewoid) [inline app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java]: You are no longer using `path`.
That's a good thing,. You can remove it.

**Review by andrewoid — APPROVED** [review]

**Comment by sschuraytz** [comment]
Drawing with charcoal tool of varying thicknesses:
image

File: app/src/main/java/com/sschuraytz/charcoaldrawing/CharcoalTool.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        bitmap = Bitmap.createBitmap(200, 200, Bitmap.Config.ARGB_8888);
+        canvas = new Canvas(bitmap);
+        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
+        //should allow user to modify
+        paint.setStrokeWidth(3);
+        radius = 35;
+        createCircleWithNoLocation();
+    }
+
+    private void createCircleWithNoLocation()
```
Comment (andrewoid) [inline app/src/main/java/com/sschuraytz/charcoaldrawing/CharcoalTool.java]: It's weird to have a method, with no arguments, just call another method with no arguments.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**